benchmarks/NonStiffBVP/nonlinear_wpi.jmd

```diff
@@ -0,0 +1,213 @@
+---
+title: Nonlinear BVP Benchmarks
+author: Michelle Zong
+TESTING 
+---
+
+This is a benchmark of MIRK methods and Shooting method on some nonlinear boundary value problems. The testing BVPs are a set of standard BVP test problems as described [here](https://archimede.uniba.it/~bvpsolvers/testsetbvpsolvers/?page_id=29)
+
+# Setup
+
+Fetch required packages.
+
+```julia
+using BoundaryValueDiffEq, OrdinaryDiffEq, DiffEqDevTools, BenchmarkTools, BVProblemLibrary, Weave, Plots
+```
+
+Set up the benchmarked solvers.
+
+```julia
+setups = [ Dict(:alg=>MIRK2(), :dt=>0.1),
+            Dict(:alg=>MIRK3(), :dt=>0.1),
+            Dict(:alg=>MIRK4(), :dt=>0.1),
+            Dict(:alg=>MIRK5(), :dt=>0.1),
+            Dict(:alg=>MIRK6(), :dt=>0.1),]
+labels = ["MIRK2";
+               "MIRK3";
+               "MIRK4";
+               "MIRK5";
+               "MIRK6"];
+```
+
+Sets tolerances.
+
+```julia
+abstols = 1.0 ./ 10.0 .^ (1:4)
+reltols = 1.0 ./ 10.0 .^ (1:4);
+```
+
+# Benchmarks
+
+We here run the BVP benchmarks for all the MIRK methods.
+
+## Nonlinear boundary value problems
+
+### Nonlinear BVP 1
+
+```julia
+prob_1 = BVProblemLibrary.prob_bvp_nonlinear_1
+sol_1 = solve(prob_1, MIRK3(), abstol=1e-8, reltol=1e-8, dt = 0.1)
+testsol_1 = TestSolution(sol_1.t, sol_1.u)
+wp_1 = WorkPrecisionSet(prob_1, abstols, reltols, setups; names = labels, print_names = true, appxsol = testsol_1, 
+                        dense=false, save_everystep=false,numruns=100,maxiters=10000000, timeseries_errors=false,verbose=false)
+plot(wp_1)
```

Comment:
Reduce the maxiters and numruns for now. We can increase them once we have some results